platform-core-module/src/test/java/com/softicar/platform/core/module/permission/UserModulePermissionCacheTest.java

```diff
@@ -0,0 +1,159 @@
+package com.softicar.platform.core.module.permission;
+
+import com.softicar.platform.core.module.AGCoreModuleInstance;
+import com.softicar.platform.core.module.access.module.AbstractModuleTest;
+import com.softicar.platform.core.module.module.instance.IModuleInstance;
+import com.softicar.platform.core.module.permission.assignment.AGModuleInstancePermissionAssignment;
+import com.softicar.platform.core.module.role.AGRole;
+import com.softicar.platform.core.module.role.permission.AGRolePermission;
+import com.softicar.platform.core.module.role.user.AGRoleUser;
+import com.softicar.platform.core.module.user.AGUser;
+import com.softicar.platform.core.module.uuid.AGUuid;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import org.junit.Test;
+
+public class UserModulePermissionCacheTest extends AbstractModuleTest {
+
+	private static final UUID A = UUID.fromString("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa");
+	private static final UUID B = UUID.fromString("bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb");
+	private static final UUID C = UUID.fromString("cccccccc-cccc-cccc-cccc-cccccccccccc");
+
+	private final AGUser alice;
+	private final UserModulePermissionCache cacheAlice;
+	private final AGUser bob;
+	private final UserModulePermissionCache cacheBob;
+	private final IModuleInstance<?> moduleInstance;
+
+	public UserModulePermissionCacheTest() {
+
+		this.alice = insertUser("alice");
+		this.cacheAlice = new UserModulePermissionCache(alice);
+		this.bob = insertUser("bob");
+		this.cacheBob = new UserModulePermissionCache(bob);
+		this.moduleInstance = AGCoreModuleInstance.getInstance();
```

Comment:
I think we need at least one test case that involves a different/second module instance, to ensure that the module instance in fact matters to the implementation of `hasModulePermission`.